binding/nodejs/index.js

```diff
@@ -0,0 +1,215 @@
+    }
+
+    const pvCheetah = require(libraryPath);
+
+    let cheetahHandleAndStatus = null;
+    try {
+      cheetahHandleAndStatus = pvCheetah.init(accessKey, modelPath, endpointDurationSec);
+    } catch (err) {
+      pvStatusToException(PV_STATUS_T[err.code], err);
+    }
+
+    const status = cheetahHandleAndStatus.status;
+    if (status !== PV_STATUS_T.SUCCESS) {
+      pvStatusToException(status, "Cheetah failed to initialize");
+    }
+
+    this._handle = cheetahHandleAndStatus.handle;
+    this._pvCheetah = pvCheetah;
+    this._sampleRate = pvCheetah.sample_rate();
+    this._frameLength = pvCheetah.frame_length();
+    this._version = pvCheetah.version();
+  }
+
+  /**
+   * @returns number of audio samples per frame (i.e. the length of the array provided to the process function)
+   * @see {@link process}
+   */
+  get frameLength() {
+    return this._frameLength;
+  }
+
+  /**
+   * @returns the audio sampling rate accepted by the process function
+   * @see {@link process}
+   */
+  get sampleRate() {
+    return this._sampleRate;
+  }
+
+  /**
+   * @returns the version of the Cheetah engine
+   */
+  get version() {
+    return this._version;
+  }
+
+  /**
+   * Processes a frame of audio and returns newly-transcribed text and a flag indicating if an endpoint has been detected.
+   * Upon detection of an endpoint, the client may invoke `.flush` to retrieve any remaining transcription.
+   *
+   * @param {Array} pcm Audio data. The audio needs to have a sample rate equal to `.sampleRate` and be 16-bit linearly-encoded.
+   * This function operates on single-channel audio. If you wish to process data in a different
+   * sample rate or format consider using `.processFile`.
+   * @returns {string, bool} Inferred transcription, and a flag indicating if an endpoint has been detected.
+   */
+  process(pcm) {
+    if (
+      this._handle === 0 ||
+      this._handle === null ||
+      this._handle === undefined
+    ) {
+      throw new PvStateError("Cheetah is not initialized");
+    }
+
+    if (pcm === undefined || pcm === null) {
+      throw new PvArgumentError(
+        `PCM array provided to process() is undefined or null`
+      );
+    } else if (pcm.length === 0) {
+      throw new PvArgumentError(`PCM array provided to process() is empty`);
+    }
+
+    let partialTranscriptAndStatus = null;
+    try {
+      partialTranscriptAndStatus = this._pvCheetah.process(this._handle, pcm);
+    } catch (err) {
+      pvStatusToException(PV_STATUS_T[err.code], err);
+    }
+
+    const status = partialTranscriptAndStatus.status;
+    if (status !== PV_STATUS_T.SUCCESS) {
+      pvStatusToException(status, "Cheetah failed to process the audio frame");
+    }
+
+    return [ partialTranscriptAndStatus.transcript, partialTranscriptAndStatus.is_endpoint === 0 ? false : true ]
+  }
+
+  /**
+   * Processes a frame of audio and returns newly-transcribed text and a flag indicating if an endpoint has been detected.
```

Comment:
I don't think this is correct description

Reply:
copy paste from leopard, but forgot to update the descriptions, good catch! :)